### PR TITLE
add support for the pgx stdlib driver

### PR DESCRIPTION
**Pull request by andy-kimball**
Add support for the pgx Postgres driver. This only supports the stdlib
driver, which integrates with the standard Go sql package. Usage of the
custom pgx driver is still not supported.

As part of supporting pgx, copyist needed to support the QueryContext,
ExecContext, and BeginTx driver functions. The tests are also refactored
in order to share common driver-specific tests.

### CODE REVIEW

File: drivertest/commontest/common.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+	// called in that case.
+	if closer != nil {
+		closer.Close()
+	}
+
+	os.Exit(code)
+}
+
+// TestQuery fetches a single customer.
+func TestQuery(t *testing.T, driverName, dataSourceName string) {
```
Comment (petermattis) [inline drivertest/commontest/common.go:91]: I was confused when initially reading this to see a `Test*` function in a non `*_test.go` file. Makes sense to me now what is going on. Not sure if there is anything to be done about that. Perhaps prefix this functions with some other word such as `DoTestQuery` or `RunTestQuery`, though that is a bit ugly. Fine to leave as-is.

File: drivertest/commontest/common.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+		Time: parseTime("2000-01-01T10:00:00+00:00", driverName), Bool: true,
+		Bytes: []byte{'A', 'B', 'C', 'D'}, Flt: 1.1, Dec: "100.1234",
+		FltArr: "{1.1,1.2345678901234567}", Uuid: []byte("8b78978b-7d8b-489e-8ca9-ac4bdc495a82"),
+	}, out)
+
+	rows.Next()
+	require.NoError(t, rows.Scan(
+		&out.Int, &out.Str, &out.TimeZ, &out.Time, &out.Bool, &out.Bytes,
+		&out.Flt, &out.Dec, &out.FltArr, &out.Uuid))
+	require.Equal(t, DataTypes{
```
Comment (petermattis) [inline drivertest/commontest/common.go:184]: Huh, I'm surprised this works. Normally you can't do comparison on structs if the struct contains a slice such as `[]byte`. Must be some magic going on in the `require` package.

**Review by petermattis — APPROVED** [review]
LGTM

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# permissions and limitations under the License.
+
+# test re-records all copyist tests from a clean state and then re-runs them
+# using copyist playback.
+# Run this before submitting a PR.
+.PHONY: test
+test:
+	# Delete all copyist files and clear the test cache to get clean slate.
+	@find . -type f -name '*.copyist' -delete
+	@go clean -testcache
```
[the remark this answers is not in the document]
Comment (andy-kimball) [reply Makefile]: Done

File: drivertest/commontest/common.go (hunk 1 above)
Comment (andy-kimball) [reply drivertest/commontest/common.go]: I switched to use `RunTestXXX`.

**Pull request merged**